### PR TITLE
Add API/JSON to delete shortlog and or calendarlog on a per idx base

### PR DESCRIPTION
**Pull request by rwaaren**
detailed history for some devices might be interresting while for other devices it will never be looked at
this API call provides the option to selectively delete the shortlog and/or calendarlog of a device without the need
to use external tools.
Like it is already available for switchtype devices and text sensors.

/json.htm?idx=<idx>&type=command&param=deletehistory&historytype=calenderlog
/json.htm?idx=<idx>&type=command&param=deletehistory&historytype=shortlog

### CODE REVIEW

File: main/WebServer.cpp [hunk 1]
```diff
@@ -496,6 +496,8 @@ namespace http {
 			RegisterCommandCode("downloadready", std::bind(&CWebServer::Cmd_DownloadReady, this, _1, _2, _3));
 			RegisterCommandCode("update_application", std::bind(&CWebServer::Cmd_UpdateApplication, this, _1, _2, _3));
 			RegisterCommandCode("deletedatapoint", std::bind(&CWebServer::Cmd_DeleteDatePoint, this, _1, _2, _3));
+			RegisterCommandCode("deletehistory", std::bind(&CWebServer::Cmd_DeleteHistory, this, _1, _2, _3));
```
Comment (neheb) [inline main/WebServer.cpp:499]: I would say avoid std::bind here.

Comment (rwaaren) [reply main/WebServer.cpp:499]: @neheb 
Thx for the review. Please understand I am not fluent in c++ 
I copied an existing function with similar functionality and only really changed the SQL's inside. 
So much obliged if you could elaborate a bit on why this std::bind is different then the ones already used and/or on the reason why this is bad coding practice. 

Always eager to learn!

Comment (neheb) [reply main/WebServer.cpp:499]: meh nvm. it's fine for this PR.

**Comment by gizmocuz** [comment]
There is already a function doing this. This is duplicate code that now needs to be modified on two locations when we add a new log table
Best to use one function to perform this task (in the original function that does this job call the new function)

Still it is very dangerous...

**Comment by rwaaren** [comment]
> There is already a function doing this. This is duplicate code that now needs to be modified on two locations when we add a new log table
> Best to use one function to perform this task (in the original function that does this job call the new function)
> 
> Still it is very dangerous...

@gizmocuz 
Sorry but I am not aware of an existing function for this task.
I only found DeleteDataPoint but considered it as too different. 
Are you referring to that one or is there another?

Thx!

**Comment by rwaaren** [comment]
Looking at DeleteDatapoint again, I might be able to create a PR to implement your proposed solution. Will have a look today/this weekend

**Comment by gizmocuz** [comment]
@rwaaren , just take the 'delete' calls for the logs from 'SQLHelper.cpp::DeleteDevices' and put them in a new function

void CSQLHelper::DeleteHistory(const std::string& idx, const bool bDeleteShortlog, const bool bDeleteDayLog)

Try to keep the internal common used variable names (idx) and please don't use const char* but const std::string& instead

**Comment by gizmocuz** [comment]
And still it's dangerous and I'm not sure plugins/scripts should be able to do this

**Comment by rwaaren** [comment]
Understand. Closing this PR and will prepare a new one.

**Comment by gizmocuz** [comment]
Before you do, why is this needed?

**Pull request closed without merging**